src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/environment/Environment.kt

```diff
@@ -0,0 +1,77 @@
+package no.skatteetaten.aurora.gobo.graphql.environment
+
+import com.expediagroup.graphql.annotations.GraphQLIgnore
+import no.skatteetaten.aurora.gobo.integration.boober.EnvironmentDeploymentRef
+import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationDeploymentResource
+import no.skatteetaten.aurora.gobo.integration.mokey.StatusCheckResource
+
+enum class EnvironmentStatusType {
+    /**
+     * No status from Phil or Mokey
+     */
+    INACTIVE,
+
+    /**
+     * Aurora configuration has been applied by Boober, but no ApplicationDeployment is registered.
+     */
+    APPLIED,
+
+    /**
+     * Failed in either Boober/Phil or Mokey for any application
+     */
+    FAILED,
+
+    /**
+     * The deployment of the environment is completed
+     */
+    COMPLETED
+}
+
+data class EnvironmentStatus(
+    val state: EnvironmentStatusType,
+    val message: String? = null,
+    val details: String? = null
+) {
+    companion object {
+        // TODO status from Phil must be added
+        fun create(ad: ApplicationDeploymentResource?) = when {
+            ad == null -> EnvironmentStatus(EnvironmentStatusType.INACTIVE)
+            ad.success() -> EnvironmentStatus(EnvironmentStatusType.COMPLETED)
+            else -> EnvironmentStatus(
+                EnvironmentStatusType.FAILED,
+                "Application not deployed",
+                "${ad.status.code} - ${ad.status.reasons.detailedStatusMessage()}"
+            )
+        }
+    }
+}
+
+data class EnvironmentApplication(
```

Comment:
Kan vi kalle klassen kun for Application?

Reply:
det kan vi, den er i kontekst av environment pakken så det bør vel være ganske tydelig uansett?